wordpress-mobile/WordPress-Android
# Implements: ViewModelSlice for SiteinfoHeaderCard

## Description: 
This PR moves the SiteInfoHeaderCard logic from MySiteViewModel to a seperate viewmodel slice - `SiteInfoHeaderCardViewModelSlice`

Also moves the unit tests for SiteInfoHeader card in `MySiteViewModelTest` to  `SiteInfoHeaderCardViewModelSliceTest` 

## 📷  SiteInfoHeaderCard
![Screenshot_20230922_173731](https://github.com/wordpress-mobile/WordPress-Android/assets/17463767/ebd8ae1b-70c9-41cd-91d9-d83498f2bc47)


## To test:
- Go to Jp/Wp app
- Go to dashboard 
- Verify that SiteInfoHeaderCard interactions are working as expected 
- Interactions to test 
   - Click on the Site name and change the name 
   - Click on the Site Icon and change Icon
   - Click on the Site url and verify that the user is navigated to the site in webview
   - Click on the Site Switcher → Select another site 
   - Verify that site switching is working as expected  

## Regression Notes
1. Potential unintended areas of impact
Site Info is not shown as expected

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and Unit testing

3. What automated tests I added (or what prevented me from doing so)
Unit testing

PR submission checklist:

UI Changes testing checklist:
